meshtastic/interfaces/ble/notifications.py

```
@@ -228,3 +252,460 @@ def _get_callback(self, characteristic: str) -> Callable[[Any, Any], None] | Non
+        legacy_log_handler: Callable[[Any, bytes | bytearray], None],
+        log_handler: Callable[[Any, bytes | bytearray], None],
+        from_num_handler: Callable[[Any, bytes], None],
+    ) -> None:
+        """Register BLE characteristic notification handlers on ``client``."""
+        self._current_legacy_log_handler = legacy_log_handler
+        self._current_log_handler = log_handler
+        self._current_from_num_handler = from_num_handler
+
+        def _safe_call(
+            handler: Callable[[Any, Any], None],
+            sender: Any,
```

Ruff (0.15.6)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `sender`

(ANN401)